pkg/rancher-desktop/main/commandServer/httpCommandServer.ts

```diff
@@ -630,6 +639,79 @@ export class HttpCommandServer {
       }
     }
   }
+
+  protected async listSnapshots(request: express.Request, response: express.Response, context: commandContext): Promise<void> {
+    const snapshots = await this.commandWorker.listSnapshots(context);
+
+    response.status(200).type('json').send(snapshots);
+  }
+
+  protected async createSnapshot(request: express.Request, response: express.Response, context: commandContext): Promise<void> {
+    try {
+      const [data, payloadError] = await serverHelper.getRequestBody(request, MAX_REQUEST_BODY_LENGTH);
+
+      if (payloadError) {
+        response.status(400).type('txt').send('The snapshot is invalid');
+      }
+
+      const snapshot = JSON.parse(data);
+
+      if (!snapshot.name) {
+        response.status(400).type('txt').send('The name field is required');
+      } else {
+        await this.commandWorker.createSnapshot(context, snapshot);
+
+        response.status(200).type('txt').send('Snapshot successfully created');
+      }
+    } catch (error: any) {
+      if (error.isSnapshotError) {
+        response.status(400).type('txt').send(error.message);
```

Comment:
Did you need an early return here?

(I'm going to stop noting these; please check the rest of the similar functions.)

Reply:
This is done to differentiate `rdctl` errors from 500 errors.